backend/capellacollab/core/exceptions.py

```diff
@@ -3,59 +3,39 @@
 
 import dataclasses
 
-import fastapi
-from fastapi import exception_handlers, status
+from fastapi import status
+
+
+@dataclasses.dataclass(kw_only=True)
+class BaseError(Exception):
+    status_code: int
+    title: str
+    reason: str
+    err_code: str | None = None
 
 
 @dataclasses.dataclass
-class ExistingDependenciesError(Exception):
+class ExistingDependenciesError(BaseError):
     entity_name: str
     entity_type: str
     dependencies: list[str]
 
-
-async def existing_dependencies_exception_handler(
-    request: fastapi.Request, exc: ExistingDependenciesError
-) -> fastapi.Response:
-    dependencies_str = ", ".join(exc.dependencies)
-    return await exception_handlers.http_exception_handler(
-        request,
-        fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": [
-                    f"The {exc.entity_type} '{exc.entity_name}' can not be deleted. Please remove the following dependencies first: {dependencies_str}"
-                ]
-            },
-        ),
-    )
+    def __post_init__(self):
+        self.status_code = status.HTTP_409_CONFLICT
+        self.title = f"{self.entity_name} cannot be deleted"
+        self.reason = f"""The {self.entity_type} '{self.entity_name}' can not be deleted.
+        Please remove the following dependencies first: {", ".join(self.dependencies)}
+        """
 
 
 @dataclasses.dataclass
-class ResourceAlreadyExistsError(Exception):
-    resource_name: str
-    identifier_name: str
-
-
-async def resource_already_exists_exception_handler(
-    request: fastapi.Request, exc: ResourceAlreadyExistsError
-):
-    return await exception_handlers.http_exception_handler(
-        request,
-        fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": f"A {exc.resource_name} with a similar {exc.identifier_name} already exists.",
-                "technical": f"{exc.identifier_name} already used",
-            },
-        ),
-    )
-
-
-def register_exceptions(app: fastapi.FastAPI):
-    app.add_exception_handler(
-        ExistingDependenciesError, existing_dependencies_exception_handler  # type: ignore[arg-type]
-    )
-    app.add_exception_handler(
-        ResourceAlreadyExistsError, resource_already_exists_exception_handler  # type: ignore[arg-type]
-    )
+class ResourceAlreadyExistsError(BaseError):
+    resource_name: str | None = None
+    identifier_name: str | None = None
+    status_code: int = status.HTTP_409_CONFLICT
+    title: str = dataclasses.field(kw_only=True)
+    reason: str = dataclasses.field(kw_only=True)
```

Comment:
```suggestion
```

What is this used for?

Reply:
I'm pulling these out. Using kw-only fields was a proposed solution to a mypy error ("Attributes without a default cannot follow attributes with one") because of the way I was trying to (probably over-) streamline the code by removing __init__ functions from the exception dataclasses. But then is a mess of some subclasses needing it in various places and it seems simpler to just go back to using __init__